### ISSUE TITLE
Code Engine: Error thrown if reporting interval already begins as kill() is called

### ISSUE THREAD

**Issue by marusshi** [karthikp-ubc/ThingsJS#14]
If _CodeEngine.kill()_ is called before the asynchronous callback in _CodeEngine.reportResources_ is executed (as a result from the call in _CodeEngine.reportTimer_), an error is thrown because the pubsub node is dead.

```
(node:9800) UnhandledPromiseRejectionWarning: Error: client disconnecting
```

**Comment by jungkumseok** [comment]
Related to #16

**Comment by jungkumseok** [comment]
This was not completely fixed, but was "handled" in ee5c4a12b3b59cf41f3f9d38a659679fbd1adbc6

To get rid of the problem completely, I think we'll need to rewrite Pubsub client to handle multiple "subscribers" (there could be more than one object sharing the same Pubsub client).

Going to close this issue for now, but we might have to reopen later
